qa/packaging/src/test/java/org/elasticsearch/packaging/test/PasswordToolsTests.java

```diff
@@ -130,6 +137,69 @@ public void test40GeneratePasswordsBootstrapAlreadySet() throws Exception {
         });
     }
 
+    /**
+     * The security index is created on startup.
+     * It can happen that even when the security index exists, we get an authentication failure as `elastic`
+     * user because the reserved realm checks the security index first.
+     * This is because we check the security index too early (just after the creation) when all shards did not get allocated yet.
+     * Hence, the call can result in an `UnavailableShardsException` and cause the authentication to fail.
+     * We retry here on authentication errors for a couple of seconds just to verify that this is not the case.
+     */
+    private <R> R retryOnAuthenticationErrors(final Callable<R> callable) throws Exception {
```

Comment:
Retrying here because the calls would have succeeded if we just waited a bit longer:

```
[2025-01-23T00:00:51,232][ERROR][o.e.x.s.a.e.ReservedRealm] failed to retrieve password hash for reserved user [elastic]
org.elasticsearch.action.UnavailableShardsException: at least one primary shard for the index [.security-7] is unavailable
[2025-01-23T00:00:51,237][INFO ][o.e.x.s.a.RealmsAuthenticator] Authentication of [elastic] was terminated by realm [reserved] - failed to authenticate user [elastic]
[2025-01-23T00:00:51,358][INFO ][o.e.c.r.a.AllocationService]  current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.security-7][0]]])." previous.health="YELLOW" reason="shards started [[.security-7][0]]"
```